drivers/video/vnc/vnc_receiver.c

```diff
@@ -321,8 +321,14 @@ int vnc_receiver(FAR struct vnc_session_s *session)
                   /* Inject the key press/release event into NX */
 
                   keyevent = (FAR struct rfb_keyevent_s *)session->inbuf;
+#ifndef CONFIG_VNCSERVER_KBD
                   vnc_key_map(session, rfb_getbe32(keyevent->key),
                               (bool)keyevent->down);
+#else
+                  /* Pass raw keycode to keyboard driver */
+
+                  session->kbdout(session, keyevent->down, keyevent->key);
```

Comment:
should we move the code into vnc_key_map

Reply:
Is it better to move keycode decode to application layer from this virtual keyboard driver ?

Reply:
Yes, if the user doesn't enable CONFIG_VNCSERVER_KBDENCODE, but it's better to move the code into vnc_key_map